pandas/core/indexes/base.py

```diff
@@ -473,7 +474,7 @@ def _simple_new(cls, values, name=None, dtype=None, **kwargs):
 
         result = object.__new__(cls)
         result._data = values
-        result.name = name
+        result._set_names([name])
```

Comment:
@jreback I wasn't sure if `_set_names` was getting called from `_simple_new`, so I made it explicit. Is this ok?

Also, we are not checking in `__new__` anymore (as you suggested).

Reply:
you shouldn't need to do this, and just leave the original code

setting ``.name`` name is a property that calls ``_set_names``